aptos-move/framework/aptos-framework/sources/AuctionHouse.move

```diff
@@ -0,0 +1,878 @@
+        token_names: vector<vector<u8>>,
+        selling_price: u64,
+        duration: u64
+    ) acquires ListingData{
+        let sender_addr = Signer::address_of(sender);
+
+        let start_time = Timestamp::now_microseconds();
+        let listing_id = GUID::id(&GUID::create(sender));
+        let listing_data = borrow_global_mut<ListingData>(sender_addr);
+        let listed_items = &mut listing_data.listed_items;
+
+        Event::emit_event<FPSInitEvent>(
+            &mut listing_data.fps_init_events,
+            FPSInitEvent { id: listing_id, amount: selling_price },
+        );
+
+        Table::add(listed_items, listing_id, ListedItem {
+            min_selling_price: selling_price,
+            buy_now_price: Option::some<u64>(selling_price),
+            duration,
+            start_time,
+            current_bid: Option::none<u64>(),
+            current_bidder: Option::none<address>(),
+            locked_tokens: Option::some(parse_token_ids(sender, token_creators, token_collection_names, token_names)),
+        })
+    }
+
+    // initializes an auction
+    // auction starts with current bidder being the seller and current bid being min_selling_price-1
+    fun initialize_auction(
+        sender: &signer,
+        token_creators: vector<address>,
+        token_collection_names: vector<vector<u8>>,
+        token_names: vector<vector<u8>>,
+        min_selling_price: u64,
+        buy_now_price: Option<u64>,
+        duration: u64
+    ) acquires ListingData {        
+        let sender_addr = Signer::address_of(sender);
+        
+        let start_time = Timestamp::now_microseconds();
+        let listing_id = GUID::id(&GUID::create(sender));
+        let listing_data = borrow_global_mut<ListingData>(sender_addr);
+        let listed_items = &mut listing_data.listed_items;
+
+        Event::emit_event<AuctionInitEvent>(
+            &mut listing_data.auction_init_events,
+            AuctionInitEvent { id: listing_id, min_selling_price: min_selling_price, duration: duration },
+        );
+
+        Table::add(listed_items, listing_id, ListedItem {
+            min_selling_price,
+            buy_now_price,
+            duration,
+            start_time,
+            current_bid:  Option::none<u64>(),
+            current_bidder:  Option::none<address>(),
+            locked_tokens: Option::some(parse_token_ids(sender, token_creators, token_collection_names, token_names)),
+        })
+    }
+
+    fun is_auction_active(start_time: u64, duration: u64): bool {
+        let current_time = Timestamp::now_microseconds();
+        current_time <= start_time + duration && current_time >= start_time
+    }
+
+    fun is_auction_complete(start_time: u64, duration: u64): bool {
+        let current_time = Timestamp::now_microseconds();
+        current_time > start_time + duration
+    }
+
+    public(script) fun buy_token_script(
+        sender: &signer,
+        seller: address,
+        guid_creation_num: u64
```

Comment:
should this be token_id and amount indicating buyer want to buy `token_id` with `amount`?